plugin/forward/forward_test.go

```diff
@@ -74,3 +79,45 @@ func TestSetTapPlugin(t *testing.T) {
 		t.Error("Unexpected order of dnstap plugins")
 	}
 }
+
+func TestAllUpstreamsDown(t *testing.T) {
+
+	s := dnstest.NewServer(func(w dns.ResponseWriter, r *dns.Msg) {
+		// timeout
+	})
+
+	s1 := dnstest.NewServer(func(w dns.ResponseWriter, r *dns.Msg) {
+		// timeout
+	})
+	p := proxy.NewProxy("TestHealthAllUpstreamsDown", s.Addr, transport.DNS)
+	p1 := proxy.NewProxy("TestHealthAllUpstreamsDown2", s1.Addr, transport.DNS)
+
+	p.GetHealthchecker().SetReadTimeout(10 * time.Millisecond)
+	p.GetHealthchecker().SetWriteTimeout(20 * time.Millisecond)
+	p.GetHealthchecker().SetDomain("example.org.")
+	p1.GetHealthchecker().SetReadTimeout(10 * time.Millisecond)
+	p1.GetHealthchecker().SetWriteTimeout(10 * time.Millisecond)
+	p1.GetHealthchecker().SetDomain("example1.org.")
```

Comment:
SetDomain() and SetWriteTimeout() are not really needed in your test, the default values should be good

Reply:
Removed